packages/payments-plugin/src/stripe/stripe.service.ts

```diff
@@ -0,0 +1,117 @@
+    ) {
+        this.stripe = new Stripe(this.options.apiKey, {
+            apiVersion: '2020-08-27',
+        });
+    }
+
+    async createPaymentIntent(ctx: RequestContext, order: Order): Promise<string | undefined> {
+        let customerId: string | undefined;
+
+        if (this.options.storeCustomersInStripe && ctx.activeUserId) {
+            customerId = await this.getStripeCustomerId(ctx, order);
+        }
+
+        const { client_secret } = await this.stripe.paymentIntents.create({
+            amount: order.totalWithTax,
+            currency: order.currencyCode.toLowerCase(),
+            customer: customerId,
+            automatic_payment_methods: {
+                enabled: true,
+            },
+            metadata: {
+                channelToken: ctx.channel.token,
+                orderId: order.id,
+                orderCode: order.code,
+            },
+        });
+
+        if (!client_secret) {
+            // This should never happen
+            Logger.warn(
+                `Payment intent creation for order ${order.code} did not return client secret`,
+                loggerCtx,
+            );
+        }
+
+        return client_secret ?? undefined;
+    }
+
+    async createRefund(paymentIntentId: string, amount: number): Promise<Stripe.Refund | Stripe.StripeError> {
+        // TODO: Consider passing the "reason" property once this feature request is addressed:
+        // https://github.com/vendure-ecommerce/vendure/issues/893
+        try {
+            const refund = await this.stripe.refunds.create({
+                payment_intent: paymentIntentId,
+                amount,
+            });
+
+            return refund;
+        } catch (e: any) {
+            return e as Stripe.StripeError;
+        }
+    }
+
+    constructEventFromPayload(payload: Buffer, signature: string): Stripe.Event {
+        return this.stripe.webhooks.constructEvent(payload, signature, this.options.webhookSigningSecret);
+    }
+
+    /**
+     * Returns the stripeCustomerId if the Customer has one. If that's not the case, queries Stripe to check
+     * if the customer is already registered, in which case it saves the id as stripeCustomerId and returns it.
+     * Otherwise, creates a new Customer record in Stripe and returns the generated id.
+     */
+    private async getStripeCustomerId(ctx: RequestContext, activeOrder: Order): Promise<string | undefined> {
+        // Load relation with customer not available in the response from activeOrderService.getOrderFromContext()
+        const order = await this.connection.getRepository(Order).findOne(activeOrder.id, {
+            relations: ['customer'],
+        });
+
+        if (!order || !order.customer) {
+            // This should never happen
+            return undefined;
+        }
+
+        const { customer } = order;
+
+        // TODO: fix typings
+        if ((customer.customFields as any).stripeCustomerId) {
```

Comment:
Spent a bit of time trying to get this to work but also to no avail!

I found this issue: https://github.com/microsoft/TypeScript/issues/46617 which _might_ be related.

Reply:
That was a good lead. I managed to get it working using the full path to `CustomerCustomFields` in both Braintree and Stripe plugins.

* `stripe/types.ts`:
```typescript
import '@vendure/core/dist/entity/custom-entity-fields';

declare module '@vendure/core/dist/entity/custom-entity-fields' {
    interface CustomCustomerFields {
        stripeCustomerId?: string;
    }
}
```

* `braintree/types.ts`:
```typescript
import '@vendure/core/dist/entity/custom-entity-fields';

// ...

declare module '@vendure/core/dist/entity/custom-entity-fields' {
    interface CustomCustomerFields {
        braintreeCustomerId?: string;
    }
}
```

Intellisense support in VSCode is a bit flaky but it works and builds fine.

Reply:
Great that you got it working! In that case, let's add a comment linking to that TS issue explaining why we are using a deep import, otherwise someone might "clean it up" in the future and break things without knowing why.